### DISCUSSION TITLE
Authentication/scan using trivy in private Maven packages

### DISCUSSION THREAD

**Discussion by w4lentim** [aquasecurity/trivy#4644] [Q&A]
### Question

Hi,
I have a question about /scan for private Maven packages.

We are running a self-hosted Artifactory instance which contains some general POMs. These are included by our Java/Maven projects as parent POMs:

<parent>
	<groupId>com.company</groupId>
	<artifactId>company-root-pom</artifactId>
	<version>1.0.0</version>
</parent>

When running `trivy fs -d .` to identify vulnerabilities in our dependencies, trivy can't resolve the parent POMs (which contain additional dependencies):

`2023-06-015T09:15:00.354Z	DEBUG	parent POM not found: 2 errors occurred:`
	`* stat /builds/my-app/pom.xml: no such file or directory`
	`* com.company:company-root-pom:1.0.0 was not found in local/remote repositories`

The scan does not fail, but it would be great if the dependencies from the parent POM would be scanned.

Anyone have any idea how to resolve this issue? is it possible to skip a specific dependency in the pom.xml file?

Thank u for your attention and feedback!

Trivy version:
`trivy version v0.41.0`

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

Table

### Mode

None

### Operating System

Ubuntu WSL

### Version

```bash
trivy v0.41.0
```

**Comment by knqyf263** [comment]
You probably need to run `mvn install`, `mvn dependency:copy-dependencies` or something like that in advance.
